### PR TITLE
Feature/api documentation

### PR DESCRIPTION
**Pull request by nequalszero**
Add Ingest and DataClient related code samples
Add custom ingest related documentation to DataClient tutorial
Add tutorial for retrieving model reports
Add DataClient#get_ingest_status to tutorial

### CODE REVIEW

File: docs/source/tutorial/data_examples.rst [hunk 1]
```diff
@@ -9,9 +9,15 @@ on Citrination. To access the data client, instantiate ``CitrinationClient`` and
...
+upload a file or a directory to a dataset on Citrination using the "default"
+ingester. This method is useful for uploading JSON files that follow the PIF
+schema, as well as files that do not require additional processing with a custom
+ingester.
+
+.. attention::
+  For files that require additional processing, see the documentation for the
+  ``upload_with_ingester`` and ``upload_with_template_csv_ingester`` sections.
 
 This method is parameterized with the following values:
```
Comment (ahl389) [inline docs/source/tutorial/data_examples.rst]: In a normal workflow is the `dataset_id` argument something they will have to write code to obtain? If it is - you might want to add a note with a link to the docs that explain how to get that ID

File: docs/source/tutorial/data_examples.rst [hunk 2]
```diff
@@ -63,6 +69,134 @@ You can also specify that a folder be renamed when it is uploaded to Citrination
...
+.. attention::
+  Note that the `IngesterList#where` method returns a new instance of `IngesterList`,
+  and requires indexing into the `IngesterList#ingesters` attribute to ultimately
+  select an ingester.
+
+Viewing an Ingester's Arguments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once you have an `Ingester`, you can check its optional and required arguments
+via the `Ingester#arguments` attribute.
```
Comment (ahl389) [inline docs/source/tutorial/data_examples.rst]: I'd suggest adding a sentence or two about why someone might take this action (other than its good info to know!) - will this help them accomplish future tasks?

File: docs/source/tutorial/data_examples.rst [hunk 3]
```diff
@@ -63,6 +69,134 @@ You can also specify that a folder be renamed when it is uploaded to Citrination
...
+Uploading Data Using a Custom Ingester
+--------------------------------------
+
+The ``upload_with_ingester`` method allows for custom ingesters to be used when
+uploading a file.
+
+This method is parameterized with the following values:
+
+* **dataset_id** - The integer value of the ID of the dataset to which
+  you will be uploading
```
Comment (ahl389) [inline docs/source/tutorial/data_examples.rst]: see earlier note about `dataset_id`

File: docs/source/tutorial/data_examples.rst [hunk 4]
```diff
@@ -63,6 +69,134 @@ You can also specify that a folder be renamed when it is uploaded to Citrination
...
+* **source_path** - The path to the file that you want to upload and for the
+  ingester to then process
+* **ingester** - The custom ingester you want to use
```
Comment (ahl389) [inline docs/source/tutorial/data_examples.rst]: id? name? do either work?

File: docs/source/tutorial/data_examples.rst (hunk 1 above)
Comment (nequalszero) [reply docs/source/tutorial/data_examples.rst]: The `dataset_id` argument is just an integer/string, you kind of just need to know which dataset you want to work with. You can't really obtain it from code unless you've just created a new dataset.

File: docs/source/tutorial/data_examples.rst (hunk 2 above)
Comment (nequalszero) [reply docs/source/tutorial/data_examples.rst]: 👍

File: docs/source/tutorial/data_examples.rst (hunk 4 above)
Comment (nequalszero) [reply docs/source/tutorial/data_examples.rst]: This is intended to be an instance of the Ingester class. I'll capitalize and accent the word `Ingester`

**Pull request merged**